subsys/net/ip/net_mgmt.c

```diff
@@ -56,7 +56,7 @@ static inline void mgmt_push_event(u32_t mgmt_event, struct net_if *iface,
 	ARG_UNUSED(length);
 #endif /* CONFIG_NET_MGMT_EVENT_INFO */
 
-	k_sem_take(&net_mgmt_lock, K_FOREVER);
+	k_sem_take(&net_mgmt_lock, k_is_in_isr() ? K_NO_WAIT : K_FOREVER);
```

Comment:
From https://github.com/zephyrproject-rtos/zephyr/pull/8027#pullrequestreview-125533551 :

What it k_sem_take() fails with K_NO_WAIT?

Reply:
Actually, I'll get rid of that patch: net mgmt is a user notifying mechanism, not a core one so it should not be used in ISR mode.